src/NuGetForUnity/Editor/NugetPreferences.cs

```diff
@@ -82,19 +87,20 @@ public override void OnGUI(string searchContext)
                     {
                         var pathCheck = NugetHelper.GetAssetsRelativePath(newPath);
 
-                        // if the path root is different or it is not under Assets folder, we issue a warning
-                        if (pathCheck == newPath || pathCheck.StartsWith(".."))
-                        {
-                            Debug.LogWarning(
-                                "Putting packages.config outside of Assets folder disables the functionality of restoring packages if the file is changed on the disk.");
-                        }
+                        // if the path root is different or it is not under Assets folder, we want to show a warning message
+                        shouldShowPackagesConfigPathWarning = pathCheck == newPath || pathCheck.StartsWith("..");
```

Comment:
If I change the path manually in the file and then open Preferences window this field will be false and I will not see the warning. You should probably initialize this field to a proper value in OnEnable().